src/core/internal/fetch/fetchClient.ts

```diff
@@ -35,15 +40,15 @@ export class FetchClient implements HttpClient {
         config?: { headers: Record<string, string> },
         data?: unknown,
     ): Promise<T> {
-        let body;
+        let body = '{}';
         if (data) {
             if (typeof data === 'object' && !this.isBuffer(data)) {
                 body = JSON.stringify(data);
                 config = config ?? { headers: {} };
                 config.headers = config.headers ?? {};
                 config.headers['Content-Type'] = 'application/json';
             } else {
-                body = data;
+                body = typeof data === 'string' ? data : JSON.stringify(data);
```

Comment:
this wont work for `Storage` API where `Buffer` or `ArrayBuffer` is passed. This, I guess, should be reverted.

Reply:
d726edc